plans/common.sh

```
@@ -0,0 +1,68 @@
+  exit 1
+fi
+# Validate SOURCE_REF is a 40-char hex SHA (prevents shell injection)
+if [[ ! "${SOURCE_REF}" =~ ^[0-9a-f]{40}$ ]]; then
+  echo "ERROR: SOURCE_REF must be a 40-char hex SHA, got: ${SOURCE_REF}" >&2
+  exit 1
+fi
+
+sysctl -w user.max_user_namespaces=15000
+
+git clone "${SOURCE_REPO}" proxy-src
+cd proxy-src
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)